py/server/deephaven/pandas.py

```diff
@@ -141,6 +182,20 @@ def to_table(df: pandas.DataFrame, cols: List[str] = None) -> Table:
             if diff_set:
                 raise DHError(message=f"columns - {list(diff_set)} not found")
 
+        # check if all the cols are pyarrow-backed and if so create an intermediate pyarrow table to upload to DH
+        df_dtypes = [df[col].dtype for col in cols]
+        if all(_is_pyarrow_backed(dt) for dt in df_dtypes):
+            pa_arrays = []
+            for col in cols:
+                pa_arrays.append(df[col].array)
+            pa_table = pa.Table.from_arrays(arrays=pa_arrays, names=cols)
+            dh_table = arrow.to_table(pa_table)
+            del pa_table
+            return dh_table
```

Comment:
https://arrow.apache.org/docs/python/pandas.html#handling-pandas-indexes
This indicates that there is a `pyarrow.Table.from_pandas()` method.  It also mentions `preserve_index` as a consideration.

Reply:
1. pyarrow 11.0.0 doesn't seem to be changed to take advantage of pandas 2.0 arrow-backend (at least explicitly) yet
2. even when a future version of pyarrow does, for many Pandas dtypes it still needs to do conversion
3. since DH arrow module is limited in the types we support, there will be cases where arrow.to_table() fails and we'll need to revert back to the non-arrow way, which is probably a bad idea in itself as it takes more time and gives the user the impression that all is well.

So overall it is still better for us to do the check upfront and select either path explicitly.

Reply:
My comment was more about why we are creating a bunch of arrow arrays to create a table instead of using more direct conversions.